### PR TITLE
fix regression functions: They ignored the reduce agrument

### PR DESCRIPTION
**Pull request by boeddeker**
- `reduce='none'` never worked in Pytorch, because `reduce` is interpreted as `bool` and `bool('none') is False`.
 - rename `reduce` to `reduction` to follow Pytorch names.
   - The breaking change is intended here (I wanted to break all code, but that does not work)

### CODE REVIEW

File: padertorch/ops/losses/regression.py [hunk 1]
```diff
@@ -9,20 +9,20 @@ def _get_scaling_factor(targets, estimates):
...
         return array
-    if reduce == 'sum':
+    if reduction == 'sum':
         return torch.sum(array)
-    elif reduce == 'mean':
+    elif reduction == 'mean':
         return torch.mean(array)
     else:
-        raise ValueError(f'Unknown reduce: {reduce}. Choose from "sum", '
+        raise ValueError(f'Unknown reduce: {reduction}. Choose from "sum", '
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:20]: Change to "Unknown reduction"

File: padertorch/ops/losses/regression.py [hunk 2]
```diff
@@ -49,16 +49,23 @@ def log_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
-        F.mse_loss(estimate, target, reduce='none').mean(dim=-1).log10(),
-        reduce=reduce
+        F.mse_loss(estimate, target, reduction='none').mean(dim=-1).log10(),
+        reduction=reduction
     )
 
 
 def sdr_loss(estimates: torch.Tensor, targets: torch.Tensor,
-             reduce: str = 'mean'):
+             reduction: str = 'mean', axis=-1):
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:68]: Call this `dim` to also follow PyTorch?

File: padertorch/ops/losses/regression.py (hunk 1 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py]: Done

File: padertorch/ops/losses/regression.py (hunk 2 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py]: Good point. Most pytorch functions support both.
But this change was not intended. I removed the axis argument.

File: padertorch/ops/losses/regression.py [hunk 3]
```diff
@@ -49,45 +49,59 @@ def log_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
+    tensor(-6.5167)
+    >>> sdr_loss(torch.tensor(estimate), torch.tensor(target), reduction=None)
+    tensor([-9.8528, -3.1806])
+
     """
     # Calculate the SNR. The square in the power computation is moved to the
     # front, thus the 20 in front of the log
     snr = 20 * torch.log10(
-        torch.norm(targets, dim=-1) / torch.norm(estimates - targets, dim=-1)
+        torch.norm(targets, dim=axis) / torch.norm(estimates - targets, dim=axis)
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:90]: Looks like `axis` is not defined here

Comment (thequilo) [reply padertorch/ops/losses/regression.py:90]: And `estimates` and `targets` are not defined

File: padertorch/ops/losses/regression.py [hunk 4]
```diff
@@ -49,45 +49,59 @@ def log_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
-        reduce=reduce
+        F.mse_loss(estimate, target, reduction='none').mean(dim=-1).log10(),
+        reduction=reduction
     )
 
 
-def sdr_loss(estimates: torch.Tensor, targets: torch.Tensor,
-             reduce: str = 'mean'):
+def sdr_loss(estimate: torch.Tensor, target: torch.Tensor,
+             reduction: str = 'mean'):
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:68]: How about naming them `input` and `target`, similar to many PyTorch loss functions? Also for the other loss functions here

File: padertorch/ops/losses/regression.py (hunk 3 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py]: Done

File: padertorch/ops/losses/regression.py (hunk 4 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py:95]: I am not sure, if `input` is a good name. The name hides a built-in.

**Comment by boeddeker** [comment]
I added `mse_loss` and removed `time_domain_loss_functions`

File: padertorch/ops/losses/regression.py (hunk 4 above)
Comment (thequilo) [reply padertorch/ops/losses/regression.py:95]: I agree, I don't like the name either. That's the reason why it's not called input already. But I thought it might be good to use the same names as PyTorch.

File: padertorch/ops/losses/regression.py [hunk 5]
```diff
@@ -3,26 +3,53 @@
...
+
+    >>> estimate = [[1., 2, 3], [4, 5, 6]]
+    >>> target = [[2., 3, 4], [4, 0, 6]]
+    >>> mse_loss(torch.tensor(estimate), torch.tensor(target))
+    tensor(0.9208)
+    >>> mse_loss(torch.tensor(estimate), torch.tensor(target), reduction=None)
+    tensor([0.0000, 0.9208])
+    """
+    return _reduce(
+        F.mse_loss(estimate, target, reduction='none').mean(dim=-1).log10(),
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:46]: `log10` shouldn't be there for `mse_loss`

File: padertorch/ops/losses/regression.py [hunk 6]
```diff
@@ -49,45 +76,59 @@ def log_mse_loss(estimate: torch.Tensor, target: torch.Tensor,
...
+    >>> log_mse_loss(torch.tensor(estimate), torch.tensor(target))
+    tensor(0.9208)
+    >>> log_mse_loss(torch.tensor(estimate), torch.tensor(target), reduction=None)
+    tensor([0.0000, 0.9208])
     """
     # Use the PyTorch implementation for MSE, should be the fastest
     return _reduce(
-        F.mse_loss(estimate, target, reduce='none').mean(dim=-1).log10(),
-        reduce=reduce
+        F.mse_loss(estimate, target, reduction='none').mean(dim=-1).log10(),
```
Comment (thequilo) [inline padertorch/ops/losses/regression.py:89]: I think the mean should be a sum to be the same as the Demystifying TasNet paper

File: padertorch/ops/losses/regression.py (hunk 5 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py]: Yes, fixed

File: padertorch/ops/losses/regression.py (hunk 6 above)
Comment (boeddeker) [reply padertorch/ops/losses/regression.py:89]: I'd like to keep the mean, here, otherwise it is not a mean square error.
It has still the same gradient as the loss denoted in the Demystifying TasNet paper.
So this change has no large impact. (Moving the 1/T outside the log it is only a bias)

I don't know how the mean operation is implemented,
but in theory the mean operation can be more stable (e.g. avoid overflow).
(Probably it relies on sum and is not more stable)

File: padertorch/ops/losses/regression.py (hunk 4 above)
Comment (jensheit) [reply padertorch/ops/losses/regression.py:95]: I think it is more important to have the same order than the same name, so I would keep the names as they are.

**Review by thequilo — APPROVED** [review]
